### ISSUE TITLE
Accessibility: Shared issues, home and prefs

### ISSUE THREAD

**Issue by lvuch** [rancher/dashboard#4388]
Master: https://github.com/rancher/dashboard/issues/4324

1.
###

### Shared issues

- [Repeats in other pages] Heading levels should only increase by one (h1, h2, h3, h4...)
- Alt tag missing in Logo images ![](RackMultipart20211014-4-19dhorn_html_cfc8187227b52dd0.png)
- Menu button (hamburger icon) is not accessible: it should be converted into a button so that it becomes keyboard navigable, and SR can identify it as a clickable element. In this same element, the &quot;hamburger icon&quot; should have a label or an aria-hidden=&quot;true&quot; accompanied by a &quot;sr-only&quot; \&lt;span\&gt; to provide a description to SR users. ![](RackMultipart20211014-4-19dhorn_html_2c4972db22c08a10.png)
- Also, the anchors within the hamburger menu need to use the &quot;href&quot; tag so that they can be valid and thus keyboard focusable.
- More options and user profile should be a button and follow the same specifications as described for the hamburger button ![](RackMultipart20211014-4-19dhorn_html_ea6e8710936e9f0f.png). Additionally, the user avatar should always contain an Alt attribute.
- \&lt;html\&gt; element is missing lang attribute
- When a new cluster is created the text is muted ![](RackMultipart20211014-4-19dhorn_html_f0c6a344b02e9741.png)and the color contrast fails
- ![](RackMultipart20211014-4-19dhorn_html_b346fd4aacf36c4b.png) Color contrast fails for both the icon and the red text. All types of banners/alerts are missing the role=alert attribute which is important for SR and other assistive technologies to prioritize the content.
 ![](RackMultipart20211014-4-19dhorn_html_ce26fcacdfabbdb7.png)

**------**

##

##

2.
### Shared issues

- Welcome to Rancher img: alt label missing ![](RackMultipart20211014-4-19dhorn_html_f77b8129e2d86640.png)
- [Repeats in other pages] Badges: color contrast failure. Same color combination as in &quot;links in info alert&quot; ![](RackMultipart20211014-4-19dhorn_html_c17aa669ae2a0935.png)
- [Repeats in other pages] Take me to cluster dropdown/select should ideally use a native &quot;select&quot; HTML selector ![](RackMultipart20211014-4-19dhorn_html_4e4d88275ad15fc8.png) . If using the custom-made component is required, it is imperative that the Aria role presentation is implemented ([more information on this](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_presentation_role)) and removed after the element is made visible. A text label is necessary for this interactive element. Contrast fails for the icon when the dropdown is disabled
- [Repeats in other pages] Radio buttons have a double label: a label within a label. The first label could be converted into a \&lt;div\&gt; so that the input and the label stay as siblings. ![](RackMultipart20211014-4-19dhorn_html_dd594a27c7767d8b.png)

### Light theme issues

- [Repeats in other pages] Links in Info alert: link fails color contrast: ![](RackMultipart20211014-4-19dhorn_html_ba1cc8debf506e51.png) ![](RackMultipart20211014-4-19dhorn_html_d63a3eb445c08c22.png)
- [Repeats in other pages] Primary button fails color contrast: ![](RackMultipart20211014-4-19dhorn_html_2331e1ce8b5327f6.png) . The text is the most important element in an element. The background of the button is allowed to fail against the background of the website as long as the text has a good contrast against its background: ![](RackMultipart20211014-4-19dhorn_html_758d531987bb5e9e.png)
- Additional suggestion: use only 1 primary button, especially when a set of 2 or more icons are next to one another. Having 1 clear CTA provides better guidance and hierarchy to users. If both icons have the same level of hierarchy, it is recommended to use secondary styles in both.
- [Repeats in other pages] Link color fails against white background. The color should be darkened ![](RackMultipart20211014-4-19dhorn_html_535290b1671b79cf.png) ![](RackMultipart20211014-4-19dhorn_html_758d531987bb5e9e.png)

###

### Dark theme issues

- [Repeats in other pages] Links over info banner &amp; pagination color contrast failure

![](RackMultipart20211014-4-19dhorn_html_3c7d9c25171031f4.png) ![](RackMultipart20211014-4-19dhorn_html_5a7d201656eafd1b.png) ![](RackMultipart20211014-4-19dhorn_html_f4a02f8a57b88f45.png)

- [Repeats in other pages] Contrast in dropdown/select failure

![](RackMultipart20211014-4-19dhorn_html_a28014379262828f.png) ![](RackMultipart20211014-4-19dhorn_html_956b3aebac77eb64.png)

**------**

3.
### Shared issues

- [Repeats in other pages] Icons in links or in buttons: ![](RackMultipart20211014-4-19dhorn_html_a0d5a2398e344018.png)if the icon is next to a label that is self explanatory, the icon is considered &quot;decorational&quot; and therefor can be hidden with no need for an additional label (using a sr-only span). However, if the icon does not contain a label, for example a X icon (for close) this needs to be hidden (aria-hidden) and add additional information in sr-only.
- [Repeats in other pages] Sections should not be included in tabbing order unless they can trigger actions ![](RackMultipart20211014-4-19dhorn_html_1575571974231fb1.png)
- Consider using toggles instead of buttons to show saved settings. Buttons should be used when an action is expected to happen on the current screen. ![](RackMultipart20211014-4-19dhorn_html_797a75bb9d50d058.png)

**Comment by lvuch** [comment]
I disagree with use of toggle as in some cases it isn't as straight forward as on/off enable/disable

**Comment by gaktive** [comment]
Closed the existing PR as old. Based on comments here, we need to relook at this.
